mathematica/ieee754_floating_point_evaluation_test.wl

```diff
@@ -96,3 +104,258 @@ Assert[
+(* ::Input::Initialization:: *)
+bits[x_Interval]:=N[Log2[Max[Abs[x]]]];
+
+
+(* ::Input::Initialization:: *)
+a2=1/2
+
+
+(* ::Input::Initialization:: *)
+a3=CorrectlyRound[1/3!]
+
+
+(* ::Input::Initialization:: *)
+a4=CorrectlyRound[1/4!]
+
+
+(* ::Input::Initialization:: *)
+a5=CorrectlyRound[1/5!]
+
+
+(* ::Input::Initialization:: *)
+h=Interval[{CorrectlyRound[-(2^-10.977``100)],CorrectlyRound[2^-10.977``100]}]
+
+
+(* ::Input::Initialization:: *)
+k=IEEEEvaluateWithAbsoluteError[h^2]
+
+
+(* ::Input::Initialization:: *)
+\[Delta]k=Abs[IEEEEvaluateWithAbsoluteError[hh^2]-hh^2/.hh->h]
+
+
+(* ::Input::Initialization:: *)
+bits[k]
+
+
+(* ::Input::Initialization:: *)
+bits[\[Delta]k]
+
+
+(* ::Input::Initialization:: *)
+k'=IEEEEvaluateWithAbsoluteError[h^3]
+
+
+(* ::Input::Initialization:: *)
+\[Delta]k'=Abs[Collect[IEEEEvaluateWithAbsoluteError[hh^3]-hh^3,hh]/.hh->h]
+
+
+(* ::Input::Initialization:: *)
+bits[k']
+
+
+(* ::Input::Initialization:: *)
+bits[\[Delta]k']
+
+
+(* ::Text::Initialization:: *)
+(*(*TODO(phl): Propagate positiveness of the lower bound.*)*)
+
+
+(* ::Input::Initialization:: *)
+S\[FivePointedStar]1=IEEEEvaluateWithAbsoluteError[a5 k]
+
+
+(* ::Input::Initialization:: *)
+\[Delta]S\[FivePointedStar]1=Abs[Collect[IEEEEvaluateWithAbsoluteError[a5 hh^2]-a5 hh^2,hh]/.hh->h]
+
+
+(* ::Input::Initialization:: *)
+bits[S\[FivePointedStar]1]
+
+
+(* ::Input::Initialization:: *)
+bits[\[Delta]S\[FivePointedStar]1]
+
+
+(* ::Input::Initialization:: *)
+S\[FivePointedStar]2=IEEEEvaluateWithAbsoluteError[S\[FivePointedStar]1-a3]
+
+
+(* ::Input::Initialization:: *)
+\[Delta]S\[FivePointedStar]2=Abs[Collect[IEEEEvaluateWithAbsoluteError[a5 hh^2-a3]-(a5 hh^2-a3),hh]/.hh->h]
+
+
+(* ::Input::Initialization:: *)
+bits[S\[FivePointedStar]2]
+
+
+(* ::Input::Initialization:: *)
+bits[\[Delta]S\[FivePointedStar]2]
+
+
+(* ::Input::Initialization:: *)
+S\[FivePointedStar]3=IEEEEvaluateWithAbsoluteError[k' S\[FivePointedStar]2]
+
+
+(* ::Input::Initialization:: *)
+\[Delta]S\[FivePointedStar]3=Abs[Collect[IEEEEvaluateWithAbsoluteError[hh^3(a5 hh^2-a3)]-hh^3(a5 hh^2-a3),hh]/.hh->h]
+
+
+(* ::Input::Initialization:: *)
+bits[S\[FivePointedStar]3]
+
+
+(* ::Input::Initialization:: *)
+bits[\[Delta]S\[FivePointedStar]3]
+
+
+(* ::Input::Initialization:: *)
+C\[FivePointedStar]1=IEEEEvaluateWithAbsoluteError[a4 k]
+
+
+(* ::Input::Initialization:: *)
+\[Delta]C\[FivePointedStar]1=Abs[Collect[IEEEEvaluateWithAbsoluteError[a4 hh^2]-a4 hh^2,hh]/.hh->h]
+
+
+(* ::Input::Initialization:: *)
+bits[C\[FivePointedStar]1]
+
+
+(* ::Input::Initialization:: *)
+bits[\[Delta]C\[FivePointedStar]1]
+
+
+(* ::Input::Initialization:: *)
+C\[FivePointedStar]2=IEEEEvaluateWithAbsoluteError[C\[FivePointedStar]1-a2]
+
+
+(* ::Text::Initialization:: *)
+(*(*TODO(phl): Because our intervals "bleed" a bit below 0 for values that should remain positive, we don't have the expected bound here and we lose at least a bit.  Error evaluation differ after this point.*)*)
```

Comment:
```suggestion
(* ::Text:: *)
(*TODO(phl): Because our intervals "bleed" a bit below 0 for values that should remain positive, we don't have the expected bound here and we lose at least a bit.  Error evaluation differ after this point.*)
```